app/models/site.rb

```diff
@@ -207,6 +207,15 @@ def self.welcome_topic_banner_cache_key(user_id)
     "show_welcome_topic_banner:#{user_id}"
   end
 
+  def self.welcome_topic_exists_and_is_not_edited?
+    Post.joins(:topic)
+      .find_by(
+        "topics.id = :topic_id AND topics.deleted_at IS NULL AND posts.post_number = 1 AND posts.version = 1 AND posts.created_at > :created_at",
+        topic_id: SiteSetting.welcome_topic_id,
+        created_at: 1.month.ago
+      ).present?
```

Comment:
If we're only concerned about the existence, it will be more efficient to just do a `SELECT 1` query instead of loading the post and checking the presence of it.

Reply:
Ya, I like that. I'll update it.

Reply:
Also I think we also need to think about memoizing the results here instead of having it execute every time.